live/sources/SocketClient.swift

```diff
@@ -64,16 +64,36 @@ class SocketClient {
             return
         }
         
+        switch stream.streamStatus {
+        case .notOpen:
+            printError(message: "Stream is not open")
+        case .opening, .writing:
+            DispatchQueue(label: "Awaiting stream setup").async {
```

Comment:
Wouldn't it be better to create a dedicated queue when initializing or using the global queue rather than creating a new one each write?